tests/util/diffraction_calibration_synthetic_data.py

```diff
@@ -0,0 +1,300 @@
+# @file: tests/util/diffraction_calibration_synthetic_data.py:
+#
+# Preparation of synthetic data for the following unit-test classes:
+#   * TestPixelDiffractionCalibration
+#   * TestGroupDiffractionCalibration
+#   * TestDiffractionCalibrationRecipe
+#
+
+import secrets
+from collections import namedtuple
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import mantid
+import numpy as np
+from mantid.api import ITableWorkspace, MatrixWorkspace
+from mantid.dataobjects import GroupingWorkspace, MaskWorkspace
+from mantid.simpleapi import (
+    ConvertUnits,
+    CreateSampleWorkspace,
+    LoadDetectorsGroupingFile,
+    LoadInstrument,
+    Rebin,
+    RebinRagged,
+    ScaleX,
+    mtd,
+)
+from snapred.backend.dao.DetectorPeak import DetectorPeak
+from snapred.backend.dao.GroupPeakList import GroupPeakList
+from snapred.backend.dao.ingredients import DiffractionCalibrationIngredients
+from snapred.backend.dao.RunConfig import RunConfig
+from snapred.backend.dao.state.FocusGroup import FocusGroup
+from snapred.backend.dao.state.InstrumentState import InstrumentState
+from snapred.backend.dao.state.PixelGroup import PixelGroup
+from snapred.meta.Config import Resource
+from util.helpers import *
+
+Peak = namedtuple("Peak", "centre sigma height")
+
+
+def random_seed(bits: int) -> int:
+    # Generate a random seed of 'bits' length.
+    return secrets.randbits(bits)
+
+
+class SyntheticData(object):
```

Comment:
Should this be a [test fixture](https://docs.pytest.org/en/6.2.x/fixture.html)?  Same to the other items in the utils module

Reply:
I left you a separate message via slack about this.  **Yes, this should be a fixture, but that doesn't make sense right now, considering our mixed `unittest` / `pytest` test environment.**